doc/dev/mgmt/msrest_to_dpg_migration.md

```diff
@@ -0,0 +1,247 @@
+```
+
+#### After (TypeSpec-Generated SDK)
+
+```python
+# ❌ This no longer works
+model = Model(additional_properties={"custom": "value"})
+
+# ✅ Use these approaches instead
+model = Model({"custom": "value"})
+# OR
+model = Model()
+model.update({"custom": "value"})
+# OR
+model = Model()
+model["custom"] = "value"
+
+print(model)  # Shows the additional properties directly
+```
+
+#### Migration steps
+
+- Remove all `additional_properties=` parameters from model constructors
+- Replace with direct dictionary syntax or `.update()` calls
+- Replace `.additional_properties` attribute access with direct dictionary access
+
+### String Representation Matches REST API
+
+**What changed**: TypeSpec-generated model string output uses `camelCase` (matching the REST API) instead of Python's `snake_case` convention used in Swagger generation.
+**What will break**:
+
+- Code that parses or matches against model string representations
+- Tests that compare string output
+
+#### Before (Swagger-Generated SDK)
+
+```python
+model = Model(type_name="example")
+print(model)  # {"type_name": "example"}
+```
+
+#### After (TypeSpec-Generated SDK)
+
+```python
+model = Model(type_name="example")
+print(model)  # {"typeName": "example"} - matches REST API format
+```
+
+#### Migration steps
+
+- Update any code that parses model string representations to expect `camelCase`
+- Update test assertions that compare against model string output
+- Consider using property access instead of string parsing where possible
+
+### Serialization and Deserialization Methods Removed
+
+**What changed**: TypeSpec-generated models no longer include explicit `serialize()` and `deserialize()` methods. Models are now inherently serializable through dictionary access, and deserialization happens automatically through the constructor.
+
+**What will break**:
+
+- Code that calls `model.serialize()` or `Model.deserialize()`
+- Custom serialization/deserialization workflows
+- Code that depends on the specific format returned by the old serialization methods
+
+#### Before (Swagger-Generated SDK)
+
+```python
+from azure.mgmt.test.models import Model
+import json
+
+# Serialization
+model = Model(name="example", value=42)
+serialized_dict = model.serialize()  # Returns dict ready for JSON
+json_string = json.dumps(serialized_dict)
+
+# Deserialization
+json_data = json.loads(json_string)
+model = Model.deserialize(json_data)  # Static method for deserialization
+print(model.name)  # "example"
+
+# Custom serialization with options
+serialized_full = model.serialize(keep_readonly=True)
+serialized_minimal = model.serialize(keep_readonly=False)
+```
+
+#### After (TypeSpec-Generated SDK)
+
+```python
+from azure.mgmt.test.models import Model
+import json
+
+# Serialization - model is already in serialized format when accessed as dictionary
+model = Model(name="example", value=42)
+
+# Method 1: Direct dictionary conversion (recommended)
+json_string = json.dumps(dict(model))
```

Comment:
Don't need as_dict? This would do a shallow copy, and fail at deeper level?

Reply:
Yeah we can use `.as_dict()` that's a good point